tests/android-diff-size.test.ts

```diff
@@ -0,0 +1,135 @@
+        github: {
+            pr: {
+                additions: 200,
+                deletions: 10  
+            },
+            issue: {
+                labels: []
+            },
+        },
+    };
+});
+
+describe("PR diff size checks", () => {
+    it("does not warn with less than 300 lins", async () => {
+        await diffSize();
+        
+        expect(dm.warn).not.toHaveBeenCalled();
+    })
+
+    it("warns with more than 300 lines", async () => {
+        dm.danger.github.pr.deletions = 101;
+
+        await diffSize();
+        
+        expect(dm.warn).toHaveBeenCalledWith("PR has more than 300 lines of code changing. Consider splitting into smaller PRs if possible.");
+    })
+
+    it("does not warns with more than 300 lines and releases label", async () => {
+        dm.danger.github.pr.deletions = 101;
+
+        dm.danger.github.issue.labels = [
+            {
+                name: 'Releases'
+            },
+        ]
+
+        await diffSize();
+        
+        expect(dm.warn).not.toHaveBeenCalled();
+    })
+
+    it("does not warns with more than 300 lines and additions to test files", async () => {
+        dm.danger.github.pr.additions = 291;
+        dm.danger.github.pr.deletions = 10;
+
+        dm.danger.git.modified_files = ["MyProject/src/test/something/myTestFile.ko"];
+        dm.danger.git.diffForFile = jest.fn();
+
+        const mockData = { 
+            added: "src/test/something/myTestFile.ko\r\n+ func someCode();\r\n+ let someOtherVar = 0;",
+            removed: "" 
+        };
+        dm.danger.git.diffForFile.mockReturnValueOnce(Promise.resolve(mockData));
+
+        await diffSize();
+        
+        expect(dm.warn).not.toHaveBeenCalled();
+    })
+
+    it("does not warns with more than 300 lines and deletions to test files", async () => {
+        dm.danger.github.pr.additions = 291;
+        dm.danger.github.pr.deletions = 10;
+
+        dm.danger.git.modified_files = ["MyProject/src/test/something/myTestFile.ko"];
+        dm.danger.git.diffForFile = jest.fn();
+
+        const mockData = { 
+            added: "",
+            removed: "src/test/something/myTestFile.ko\r\n- func someCode();\r\n- let someOtherVar = 0;" 
+        };
+        dm.danger.git.diffForFile.mockReturnValueOnce(Promise.resolve(mockData));
+
+        await diffSize();
+        
+        expect(dm.warn).not.toHaveBeenCalled();
+    })
+
+    it("does not warns with more than 300 lines and changes to test files", async () => {
+        dm.danger.github.pr.additions = 291;
+        dm.danger.github.pr.deletions = 12;
+
+        dm.danger.git.modified_files = ["MyProject/src/test/something/myTestFile.ko"];
+        dm.danger.git.diffForFile = jest.fn();
+
+        const mockData = { 
+            added: "src/test/something/myTestFile.ko\r\n+ func someNewCode();\r\n- let someOtherNewVar = 0;",
+            removed: "src/test/something/myTestFile.ko\r\n- func someCode();\r\n- let someOtherVar = 0;" 
+        };
```

Comment:
Are we sure that's the format of the structure returned by Danger and the git gem?

I'm confused as to what changes would go in `added:` and which ones would go in `removed:` keys in those cases. For example here in `added` you have a diff that contains both `+` and `-` lines, so isn't that a mix of both an addition and a deletion, and as such why would it be in `added:` rather than `removed:`

I am not sure what the `dm.danger.git.diffForFile` function actually returns (when not mocked) in those kind of cases, and how it differently represent new files vs deleted files vs modified files that contains only deleted lines, vs modified files that contains only added lines, vs modified files that might contain a mix of additions and deletions… but I think it's worth double-checking here (otherwise this test wouldn't actually test the actual case we'd encounter in real life)

Reply:
Oh, right! That's a typo. There should be only `+` in `added` and only `-` in `removed`.

Reply:
Ah! Makes more sense now 👍 

I dug a little and found [the definition GitDSL from Danger](https://github.com/danger/danger-js/blob/main/source/dsl/GitDSL.ts#L7-L18), and it seems like `added` and `removed` are guaranteed to only contain added and removed lines. Which means that we could probably just count all the lines in those strings, without having to filter by `+` or `-`. In other words, you should be able to get rid of [the `filter(…)` calls on lines 20-21](https://github.com/Automattic/peril-settings/pull/80/files#diff-4ad737252bc200d7661627f749fe3cac6d9023a8ea69a835236657fb40c29bf9R20-R21) and avoid useless CPI cycles filtering those lines 😉

Reply:
Yup! I initially did it. But during my debug, I found that sometimes the file name doesn't get parsed out and so it can be counted as an additional line if we don't apply a proper filtering. I believe this is a bug (it doesn't happen always and it's different from the documentation), so it's possible that it's fixed in a newer version of Danger, but as per my comment above, we are using a quite older version.

Reply:
Ah, thanks for the explanation; good workaround then 👍